src/Native/CMakeLists.txt

```diff
@@ -126,22 +126,20 @@ if(CLR_CMAKE_PLATFORM_ARCH_AMD64)
   add_definitions(-D_TARGET_AMD64_=1)
   add_definitions(-D_AMD64_)
   add_definitions(-DAMD64)
-  add_definitions(-DTARGET_X64)
-  add_definitions(-DTARGET_AMD64)
-  add_definitions(-D_WIN64=1)
+  if(WIN32)
+    add_definitions(-D_WIN64=1)
+  endif()
 elseif(CLR_CMAKE_PLATFORM_ARCH_I386)
   add_definitions(-D_TARGET_X86_=1)
   add_definitions(-D_X86_)
-  add_definitions(-DTARGET_X86)
-  add_definitions(-D_WIN32=1)
+  if(WIN32)
```

Comment:
And to stop defining _WIN64 on Unix, it needs to be replaced in the sources with BIT64 as I have mentioned in other comment. The tests are failing on Unix in CI because of this.